common/v2/services/ApiService/TokenInfo/TokenInfo.ts

```diff
@@ -0,0 +1,35 @@
+import { AxiosInstance } from 'axios';
+
+import { default as ApiService } from '../ApiService';
+
+let instantiated: boolean = false;
+
+const apiUrl = 'https://api.mycryptoapi.com/tokens';
+
+export default class TokenInfoService {
+  public static instance = new TokenInfoService();
+
+  private service: AxiosInstance = ApiService.generateInstance({
+    baseURL: apiUrl
+  });
+
+  constructor() {
+    if (instantiated) {
+      throw new Error(`TokenInfoService has already been instantiated.`);
+    } else {
+      instantiated = true;
+    }
+  }
+
+  public getTokensInfo = async (contractAddresses: string[]) => {
```

Comment:
TODO comment to complete later?

Reply:
Not sure what you mean here. This service is fetching token info and is complete.